### PR TITLE
TMDM-10719: Display format on the grid list is ignored for a date field part of a complex type

### PR DESCRIPTION
**Pull request by zfkang-talend**
**What is the current behavior?** (You can also link to an open issue here)
have no format the data on view update report of the journal compare panel if set the format display.

**What is the new behavior?**
add the function to format the value in view update report on journal compare panel.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java [hunk 1]
```diff
@@ -11,26 +11,41 @@
...
+import org.dom4j.Namespace;
+import org.dom4j.QName;
 import org.dom4j.io.SAXReader;
 import org.talend.mdm.commmon.util.webapp.XSystemObjects;
+import org.talend.mdm.webapp.base.client.model.DataTypeConstants;
 import org.talend.mdm.webapp.base.server.util.Constants;
+import org.talend.mdm.webapp.base.server.util.DateUtil;
+import org.talend.mdm.webapp.base.shared.EntityModel;
+import org.talend.mdm.webapp.base.shared.TypeModel;
+import org.talend.mdm.webapp.browserecords.server.bizhelpers.DataModelHelper;
```
Comment (qcui) [inline org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java]: There are some unnecessary class import

**Comment by qcui** [comment]
There are some unnecessary class import

**Comment by qcui** [comment]
LGTM

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java [hunk 2]
```diff
@@ -151,6 +166,66 @@ public JournalTreeModel getDetailTreeModel(String[] idsArr) throws Exception {
         return root;
     }
 
+    private String formateValue(EntityModel entityModel, Map<String, String[]> formatMap, String path, String oldValue) {
```
Comment (pbabin) [inline org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java]: typo

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java [hunk 3]
```diff
@@ -151,6 +166,66 @@ public JournalTreeModel getDetailTreeModel(String[] idsArr) throws Exception {
...
+                            sdf = new SimpleDateFormat(DateUtil.DATE_TIME_FORMAT, java.util.Locale.ENGLISH);
+                        }
+
+                        try {
+                            Date date = sdf.parse(oldValue.trim());
+                            Calendar calendar = Calendar.getInstance();
+                            calendar.setTime(date);
+                            formatValue = com.amalto.webapp.core.util.Util.formatDate(value[0], calendar);
+                        } catch (Exception e) {
+                            Log.info("format has error"); //$NON-NLS-1$
```
Comment (pbabin) [inline org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java]: Why is this an info message?

File: org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java [hunk 4]
```diff
@@ -151,6 +166,66 @@ public JournalTreeModel getDetailTreeModel(String[] idsArr) throws Exception {
...
+                        }
+                    } else if (CommonUtil.numberTypeNames.contains(tm.getType().getBaseTypeName())) {
+                        try {
+                            NumberFormat nf = NumberFormat.getInstance();
+                            Number num = nf.parse(oldValue.trim());
+                            formatValue = ""; //$NON-NLS-1$
```
Comment (pbabin) [inline org.talend.mdm.webapp.journal/src/main/java/org/talend/mdm/webapp/journal/server/service/JournalDBService.java]: No need to create an empty String instance

**Pull request merged**